src/main/java/com/okta/tools/OktaAwsCliAssumeRole.java

```diff
@@ -265,6 +282,36 @@ private LoginResult logInToOkta(String username, String password) throws JSONExc
         }
     }
 
+    private LoginResult reuseSession() throws  JSONException, IOException {
+        CookieStore cookieStore = new BasicCookieStore();
+        Properties loadedProperties = new Properties();
+        loadedProperties.load(new FileReader(getCookies().toFile()));
+        loadedProperties.entrySet().stream()
+                .map(entry -> {
+                    BasicClientCookie basicClientCookie = new BasicClientCookie(entry.getKey().toString(), entry.getValue().toString());
+                    basicClientCookie
+                            .setDomain(oktaOrg);
+                    return basicClientCookie;
+                })
+                .forEach(cookieStore::addCookie);
+        if (cookieStore.getCookies().isEmpty()) {
+            return new LoginResult(null,null);
+        }
+        HttpPost httpPost = new HttpPost("https://" + oktaOrg + "/api/v1/sessions/me/lifecycle/refresh");
+        httpPost.addHeader("Accept","application/json");
+        httpPost.addHeader("Content-Type","application/json");
+        httpPost.addHeader("Cookie","sid=" + cookieStore.getCookies().get(2).getValue());
+
+        try (CloseableHttpClient httpClient = HttpClients.createSystem()) {
+            CloseableHttpResponse authnResponse = httpClient.execute(httpPost);
+
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream(65536);
+            authnResponse.getEntity().writeTo(byteArrayOutputStream);
```

Comment:
:art: Apache HTTP Components has a utility class for this kind of entity handling:

```java
EntityUtils.toString(authnResponse.getEntity())
```